### PR TITLE
Make jobMinDuration and jobMaxDuration required

### LINKED ISSUE #263: Make `jobMinDuration` and `jobMaxDuration` mandatory
After some discussion in #262, we've decided that it's better to transform `jobMinDuration` and `jobMaxDuration` into mandatory fields in the offer's `new` endpoint. This change better reflects the behaviour shown in the FE and simplifies things.

Additionally, we can think about changing the meaning of these fields a little. It probably makes more sense to have a free integer up to 12 months (not included) and then have a `12+`, just like it's shown in the FE.

### PR DESCRIPTION
**Pull request by marco-vb**
Closes #263

### CODE REVIEW

File: test/end-to-end/company.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ describe("Company endpoint", () => {
         description: "For Testing Purposes",
         contacts: ["geral@niaefeup.pt", "229417766"],
         jobType: "SUMMER INTERNSHIP",
+        jobMinDuration: 1,
+        jobMaxDuration: 6,
```
Comment (Naapperas) [inline test/end-to-end/company.js:43]: This applies to the rest of the lines that are like these: when the tests are refactored (see #169 ) these should be put in the offer creation utility.

**Review by Naapperas — CHANGES_REQUESTED** [review]
This looks good. The only thing needed is to pick better commit names

**Review by Naapperas — APPROVED** [review]

**Comment by BrunoRosendo** [comment]
@marco-vb Please update the API documentation with this change. It is stored under `docs/` and the syntax resembles markdown, you can read the docs (in the README) to know how to use it.

**Review by BrunoRosendo — CHANGES_REQUESTED** [review]
@marco-vb Please update the API documentation with this change. It is stored under docs/ and the syntax resembles markdown, you can read the docs (in the README) to know how to use it.

File: documentation/docs/offers/create.md [hunk 2]
```diff
@@ -93,23 +93,23 @@ The date when the offer will be "hidden" from the application (will be invisible
 
 <Highlight level="info">Body Parameter</Highlight>
 
-<Highlight level="success" inline>Optional</Highlight>
+<Highlight level="danger" inline>Required</Highlight>
 <Highlight level="secondary" inline>Number</Highlight>
 
 The duration of the work itself (in months).
```
Comment (Naapperas) [inline documentation/docs/offers/create.md:100]: Should be marked as required, as you have done for `jobMaxDuration`

**Review by Naapperas — CHANGES_REQUESTED** [review]

**Review by Naapperas — APPROVED** [review]
LGTM

**Review by BrunoRosendo — APPROVED** [review]
Looks good!

**Pull request merged**